### PR TITLE
Get cIRI database path from configuration variables

### LINKED ISSUE #132: Get cIRI database path from configuration variables

### PR DESCRIPTION
**Pull request by wusyong**
Fixes #132 
- Add `db_path`

# Test Plan:
CI must pass

### CODE REVIEW

File: consensus/conf.bzl [hunk 1]
```diff
@@ -1,4 +1,5 @@
 CONSENSUS_MAINNET_VARIABLES = [
+    "DB_PATH='\"ciri/ciri.db\"'",
```
Comment (thibault-martinez) [inline consensus/conf.bzl]: That's really great, now that you're at it, can you make it `ciri/ciri-mainnet.db` ?

File: consensus/conf.bzl [hunk 2]
```diff
@@ -7,6 +8,7 @@ CONSENSUS_MAINNET_VARIABLES = [
 ]
 
 CONSENSUS_TESTNET_VARIABLES = [
+    "DB_PATH='\"ciri/ciri.db\"'",
```
Comment (thibault-martinez) [inline consensus/conf.bzl]: And this one `ciri/ciri-testnet.db` ?

File: ciri/usage.h [hunk 3]
```diff
@@ -95,6 +96,7 @@ static struct cli_argument_s {
      "Limits how many milestones behind the current one the random walk can "
      "start.",
      REQUIRED_ARG},
+    {"db-path", CLI_ARG_DB_PATH, "Path to the database file.", REQUIRED_ARG},
```
Comment (thibault-martinez) [inline ciri/usage.h]: Maybe add it a short option `-d` instead of `CLI_ARG_DB_PATH `

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Comment by thibault-martinez** [comment]
@wusyong Thank you for this really great PR !
A general comment, could you make this option reside under the cIRI conf instead of the consensus conf ?

File: consensus/conf.bzl (hunk 2 above)
Comment (wusyong) [reply consensus/conf.bzl]: Should I also duplicate a database file for this one?

Comment (thibault-martinez) [reply consensus/conf.bzl]: Yes, remove the old one, create the two new and update the paths.

```
rm ciri/ciri.db
mkdir ciri/db
sqlite3 ciri/db/ciri-mainnet.db < common/storage/sql/schema.sql
sqlite3 ciri/db/ciri-testnet.db < common/storage/sql/schema.sql
```


This is temporary as we want bazel to do that for us in the future.

File: ciri/usage.h [hunk 4]
```diff
@@ -98,7 +99,7 @@ static struct cli_argument_s {
 
     {NULL, 0, NULL, NO_ARG}};
 
-static char* short_options = "hl:n:t:u:p:";
+static char* short_options = "hld:n:t:u:p:";
```
Comment (thibault-martinez) [inline ciri/usage.h]: You are missing a `:` after `l`

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Comment by thibault-martinez** [comment]
Why are the initial databases files 44k ? Are they really empty, containing only the schema ?
I think they should be 20480 bytes.

**Comment by wusyong** [comment]
Sorry, didn't aware they were modified when testing.

**Review by thibault-martinez — APPROVED** [review]

**Comment by thibault-martinez** [comment]
Thanks @wusyong this is very useful. Good first PR !

File: ciri/BUILD [hunk 5]
```diff
@@ -1,10 +1,26 @@
...
+    name = "testnet",
+    values = {"define": "network=testnet"},
+)
+
 cc_binary(
     name = "ciri",
     srcs = ["main.c"],
     copts = ["-DLOGGER_ENABLE"],
-    data = ["ciri.db"],
+    data = [
```
Comment (th0br0) [inline ciri/BUILD]: Please use a `select({":testnet": ["db/ciri-testnet.db"], ":mainnet": ["db/ciri-mainnet.db"]})` here, no need to package both simultaneously.

File: ciri/BUILD [hunk 6]
```diff
@@ -1,10 +1,26 @@
...
+
+config_setting(
+    name = "mainnet",
+    values = {"define": "network=mainnet"},
+)
+
+config_setting(
+    name = "testnet",
+    values = {"define": "network=testnet"},
+)
```
Comment (th0br0) [inline ciri/BUILD]: Do not create new  `config_setting` entries for this.
Instead, we should move the generic `config_setting` flags to, e.g., `//config/BUILD` and then reuse them here.
(e.g. right now, we already have `//gossip:mainnet` and `//gossip:testnet` that describe this parameter)

**Review by th0br0 — CHANGES_REQUESTED** [review]
Just some nits for the buildfiles :)

File: ciri/BUILD (hunk 6 above)
Comment (wusyong) [reply ciri/BUILD]: `//consensus` also has same `config_setting` entries. Should I modify them all together?

Comment (th0br0) [reply ciri/BUILD]: That'd be awesome :)

**Review by th0br0 — APPROVED** [review]

**Pull request merged**